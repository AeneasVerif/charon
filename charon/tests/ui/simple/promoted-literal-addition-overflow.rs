//@ charon-args=--mir_optimized
#[allow(arithmetic_overflow)]
fn overflow() -> &'static u32 {
    &(u32::MAX + 1)
}
