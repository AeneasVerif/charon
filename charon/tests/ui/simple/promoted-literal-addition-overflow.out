# Final LLBC before serialization:

// Full name: core::num::{u32}::MAX
pub fn MAX() -> u32

// Full name: core::num::{u32}::MAX
pub const MAX: u32 = MAX()

// Full name: test_crate::overflow
fn overflow() -> &'static (u32)
{
    let @0: &'_ (u32); // return
    let @1: &'_ (u32); // anonymous local
    let @2: u32; // anonymous local
    let @3: &'_ (u32); // anonymous local
    let @4: &'_ (u32); // anonymous local
    let @5: u32; // anonymous local
    let @6: u32; // anonymous local
    let @7: u32; // anonymous local
    let @8: u32; // anonymous local

    storage_live(@2)
    storage_live(@3)
    storage_live(@4)
    storage_live(@5)
    storage_live(@6)
    storage_live(@7)
    storage_live(@8)
    storage_live(@1)
    @7 := MAX
    @2 := move (@7) panic.+ const (1 : u32)
    @8 := MAX
    @6 := move (@8) wrap.+ const (1 : u32)
    @5 := move (@6)
    @4 := &@5
    @3 := move (@4)
    @1 := &*(@3)
    @0 := &*(@1)
    storage_dead(@1)
    return
}



